test/support/structs/person.rb

```diff
@@ -2,15 +2,16 @@
 
 require_relative "job"
 require_relative "../enums/ruby_rank"
+require_relative "../enums/diamond_rank"
 
 class Person < T::Struct
   include ActsAsComparable
 
   const :name, String
   const :age, Integer
-  const :ruby_rank, RubyRank
+  const :stone_rank, T.any(RubyRank, DiamondRank)
```

Comment:
let me know if you prefer what you had as ruby_rank and I can just create a different think rather than piggy-backing on what you had

Reply:
Cool with piggy backing on this struct 👍🏻